felix/dataplane/linux/bpf_ep_mgr.go

```diff
@@ -954,7 +1267,45 @@ func (m *bpfEndpointManager) applyProgramsToDirtyDataInterfaces() {
 				"Ignoring interface that doesn't match the host data/l3 interface regex")
 			return nil
 		}
-		if !m.ifaceIsUp(iface) {
+
+		var (
+			err                           error
+			up                            bool
+			xdpIdx, ingressIdx, egressIdx int
+		)
+
+		m.ifacesLock.Lock()
+		defer m.ifacesLock.Unlock()
+		m.withIface(iface, func(iface *bpfInterface) bool {
+			up = iface.info.ifaceIsUp()
+
+			if xdpIdx = iface.dpState.policyIdx[hook.XDP]; xdpIdx == -1 {
+				if xdpIdx, err = m.xdpPolicyMapAlloc.Get(); err != nil {
+					return false
+				}
+			}
+			if ingressIdx = iface.dpState.policyIdx[hook.Ingress]; ingressIdx == -1 {
+				if ingressIdx, err = m.policyMapAlloc.Get(); err != nil {
+					return false
+				}
+			}
+			if egressIdx = iface.dpState.policyIdx[hook.Egress]; egressIdx == -1 {
+				if egressIdx, err = m.policyMapAlloc.Get(); err != nil {
+					return false
+				}
+			}
+
+			iface.dpState.policyIdx[hook.XDP] = xdpIdx
+			iface.dpState.policyIdx[hook.Ingress] = ingressIdx
+			iface.dpState.policyIdx[hook.Egress] = egressIdx
+			return false
+		})
+
+		if err != nil {
+			return err
```

Comment:
Just spotted, this actually causes a panic (`Iter()` doesn't know what to do with an arbitrary `error`), is that what you indended?

Reply:
(I probably should have made the iter function return an enum, not an error)

Reply:
Ohhh yeah, that error is not propagated, that needs to be fixed! Thanks!